lib/rucio/core/replica.py

```diff
@@ -799,6 +808,44 @@ def _list_replicas_for_files(file_clause, state_clause, files, rse_clause, updat
             {'scope': scope, 'name': name} in files and files.remove({'scope': scope, 'name': name})
 
 
+def get_vp_endpoint():
+    vp_endpoint = config_get('virtual_placement', 'vp_endpoint', default='')
+    return vp_endpoint
+
+
+def get_multi_cache_prefix(cache_site, filename, logger=logging.log):
+
+    vp_endpoint = get_vp_endpoint()
+    if not vp_endpoint:
+        return ''
+
+    x_caches = REGION.get('CacheSites')
+    if x_caches is NO_VALUE:
+        try:
+            response = requests.get('{}/serverRanges'.format(vp_endpoint), verify=False)
+            if response.ok:
+                x_caches = response.json()
+                REGION.set('CacheSites', x_caches)
+            else:
+                REGION.set('CacheSites', {'could not reload': ''})
+                return ''
+        except requests.exceptions.RequestException as re:
+            REGION.set('CacheSites', {'could not reload': ''})
+            logger(logging.WARNING, 'In get_multi_cache_prefix, could not access {}. Excaption:{}'.format(vp_endpoint, re))
+            return ''
+
+    if cache_site not in x_caches:
+        return ''
+
+    xcache_site = x_caches[cache_site]
+    h = float(
+        unpack('Q', sha256(filename.encode('utf-8')).digest()[:8])[0]) / 2**64
```

Comment:
what does this piece of magic do?

Reply:
:) it creates a hash from the filename and maps it to [0,1) range.